### ISSUE TITLE
[Bug] Handymen do not water plants if placed in middle of benches

### ISSUE THREAD

**Issue by Amiga60077** [CorsixTH/CorsixTH#1738]
If you place plants in middle of Benches, the Handymen will not water them in CorsixTH, meaning they will wither and die because of this.

In the original Theme Hospital, Handymen leaned over the benches to watch the plants like so..

![themehospital](https://user-images.githubusercontent.com/15225746/96468317-dca6b400-1223-11eb-96a5-f0838fe318ce.PNG)

**Comment by Alberth289346** [comment]
Ha, interesting :)

**Comment by lewri** [comment]
Guess this is to do with our pathfinding. Logically the handyman can't find a path in Corsix because a bench blocks the path.
This makes sense. But realistically you could also lean over a bench in real life to water a plant. Even though that isn't practical.

To fix this we'd need to treat benches as objects that can be... walkable? You can sit on them so hopefully it isn't too much of a headache.

**Comment by Amiga60077** [comment]
Perhaps just add it to handymen as an additional feature "walk over benches" - but deny it for other patients and staff of course heh

**Comment by Alberth289346** [comment]
You don't actually need to be able to reach the plant, just stand next to it. That is, reach one of the 4 tiles directly next to it.
That should at least work for radiators.

An empty bench should perhaps be treated like a radiator, just blocking one exit/entry direction of a tile. That could also create options for #1705

**Comment by Alberth289346** [comment]
@TheCycoONE Does this count as an issue for pathfinder blocking problems?

Don't know if we have an epic issue for that, but it may be a good idea.

**Comment by driverpt** [comment]
🤔 , i think in the original game you can't even place objects that are not reachable.

**Comment by Amiga60077** [comment]
No, That photo is what I took myself of him leaning over the bench to water the plant, so I know it works in Theme Hospital as that is proof it works, I could always upload a video somewhere if you really need to see it.

**Comment by lewri** [comment]
I wanted to add an update to this -- in CorsixTH the handyman can reach over a bin to water a plant. So it appears we do have something coded to allow this to happen easily.

**Comment by TheCycoONE** [comment]
A bin isn't blocking, benches and radiators are.

**Comment by Matroftt** [comment]
> To fix this we'd need to treat benches as objects that can be... walkable? You can sit on them so hopefully it isn't too much of a headache.

Made some tests today, i changed "bench tile" flag in `bench.lua` from `complete_cell` to `only_passable`. This way handymen can lean over the benches, but unfortunately not only them but anyone can walk through benches too this way. Maybe there should be new flag for object as `leanable` or some sort? Which would work as complete cell, but allow handyman to pass through only when their task is watering and plant they need to water is near that bench

Also, when handyman "lean" over benches bench sprite covers handyman.

**Comment by ARGAMX** [comment]
> Made some tests today, i changed "bench tile" flag in `bench.lua` from `complete_cell` to `only_passable`. This way handymen can lean over the benches, but unfortunately not only them but anyone can walk through benches too this way. 

May be it should be a sort of `deadend` type - when humanoid can step on it, but can not move any direction from it except back.
Seems this is not implemented in game yet.

**Comment by Alberth289346** [comment]
> May be it should be a sort of deadend type

You likely only want this if it is the handyman, and if they are watering the plant, and the plant is behind a bench and there is no other way to reach it. (Not sure about the last part of "no other way".)

If you add such conditions, even with a deadend notion you at least need to tell the path finder that your goal is to water the plant. Otherwise, the pathfinder will always consider using it with all kinds of risks of humanoids that are not supposed to go there.

If you need to tell the pathfinder anyway, it is likely equal or less effort to make a special case for this one exception without the general new deadend concept.

**Comment by ARGAMX** [comment]
> > May be it should be a sort of deadend type
> 
> You likely only want this if it is the handyman, and if they are watering the plant, and the plant is behind a bench and there is no other way to reach it. (Not sure about the last part of "no other way".)
> 
> If you add such conditions, even with a deadend notion you at least need to tell the path finder that your goal is to water the plant. Otherwise, the pathfinder will always consider using it with all kinds of risks of humanoids that are not supposed to go there.
> 
> If you need to tell the pathfinder anyway, it is likely equal or less effort to make a special case for this one exception without the general new deadend concept.

I think that there is no need to make so many restrictions since in the original TH any humanoid can walk to or stand on bench tile and there are no any problems.

This strange logic that bench tile is not 'stepable' exclusively CTH feature.

That's why, I believe, this bug with plants watering exists in CTH.

Thats why I think `deadend` type is would be enough.

vanilla TH screenshot:

<img width="205" alt="Image" src="https://github.com/user-attachments/assets/1a1e0f16-b1aa-48f2-879f-5ec1648982f7" />

<img width="202" height="194" alt="Image" src="https://github.com/user-attachments/assets/3d3981d2-df91-4b33-bf9c-0b7c0a2977dd" />

**Comment by Alberth289346** [comment]
Fun!

Why not make the tile available then for general use? At least I'd guess that the tile is now now blocked for normal use.

**Comment by TheCycoONE** [comment]
I don't think any pathfinding changes are needed except to pathfind to the tiles around the plant instead of the plant as long as they aren't walls.

**Comment by Alberth289346** [comment]
Path finding already looks for a tile next to the plant.

Making a plant unreachable is tricky and can only be done because the blocked area check is buggy. If you disable the check or exploit the bug, and change the bench foot-print to
``` diff
diff --git a/CorsixTH/Lua/objects/bench.lua b/CorsixTH/Lua/objects/bench.lua
index c617a8b4..5f3d0615 100644
--- a/CorsixTH/Lua/objects/bench.lua
+++ b/CorsixTH/Lua/objects/bench.lua
@@ -126,19 +126,19 @@ object.usage_animations = {
 object.orientations = {
   north = {
     render_attach_position = { {0, 0}, {-1, 0}, {0, -1} },
-    footprint = { {0, 0, complete_cell = true}, {0, -1, only_passable = true, invisible = true, shareable = true} }
,
+    footprint = { {0, 0, only_side = true}, {0, -1, only_passable = true, invisible = true, shareable = t
rue} },
     use_position = "passable",
   },
   east = {
-    footprint = { {0, 0, complete_cell = true}, {1, 0, only_passable = true, invisible = true, shareable = true} },

+    footprint = { {0, 0, only_side = true}, {1, 0, only_passable = true, invisible = true, shareable = tr
ue} },
     use_position = "passable",
   },
   south = {
-    footprint = { {0, 0, complete_cell = true}, {0, 1, only_passable = true, invisible = true, shareable = true} },

+    footprint = { {0, 0, only_side = true}, {0, 1, only_passable = true, invisible = true, shareable = tr
ue} },
     use_position = "passable",
   },
   west = {
-    footprint = { {0, 0, complete_cell = true}, {-1, 0, only_passable = true, invisible = true, shareable = true} }
,
+    footprint = { {0, 0, only_side = true}, {-1, 0, only_passable = true, invisible = true, shareable = t
rue} },
     use_position = "passable",
   },
 }
```
That works. I tried `need_side` too, but that doesn't work.

However I am not sure what the `only_passable` tile is doing here. (My guess is that the use action needs it, although afaik there is no special animation to walk to a bench.)

**Comment by TheCycoONE** [comment]
That looks like the right cell flags based on the screen shot. The passable prevents building in front of it.

**Comment by ARGAMX** [comment]
> If you change the bench foot-print to
> only_side = true
> That works.

Handyman can water plants through such benches.
Is there any disadvantages of this solution?

**Comment by ARGAMX** [comment]
Theme Hospital

<img width="405" height="287" alt="Image" src="https://github.com/user-attachments/assets/33829df4-f67e-4036-a205-a378f238ffd7" />

But these benches can't be used in Theme Hospital. They unaccessible.

CorsixTH 0.69

<img width="537" height="316" alt="Image" src="https://github.com/user-attachments/assets/3ec13540-6b96-40a2-8dd6-894d7eb9be19" />

**Comment by Alberth289346** [comment]
That is a different problem, namely having the additional tile in front of the bench.

I remember seeing in another issue that patients can stand in front of the bench in original TH. In CTH that is not possible afaik.
Pretty likely that is related to benches being unusable when facing a wall.
